### PR TITLE
Upstream material correction in the ECal

### PR DESCRIPTION
**Pull request by zaborowska**
Addresses # .

Changes proposed in this pull-request:

- split cryostat volume into in front, behind, on sides
- possibility to check any of those as sensitive
- a new algorithm that writes out the dependence of energy deposited in ecal layers on the energy deposited in cryostat

To be done:

- [x] calculate maximum eta of endcaps (1.8 at the moment)
- [x] documentation

### CODE REVIEW

File: Detector/DetStudies/src/components/UpstreamMaterial.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     } else {
+       sumEupstream += hit.core().energy;
+     }
+   }
+  for(uint i = 0; i < m_numLayers; i++) {
+    sumEcells[i] /= m_samplingFraction[i];
+    integralEcells += sumEcells[i];
+    m_cellEnergyPhi[i]->Fill(phi, integralEcells);
+    m_upstreamEnergyCellEnergy2D[i]->Fill(integralEcells, sumEupstream);
+    verbose() << "total E in cell "<< i << " = " << sumEcells[i] <<"\t up to current cell = " << integralEcells << "\t upstream = " << sumEupstream << endmsg;
```
Comment (faltovaj) [inline Detector/DetStudies/src/components/UpstreamMaterial.cpp]: "total E in cell " -> "total E in layer "

File: Detector/DetStudies/tests/options/upstreamMaterial_inclinedEcal.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    OutputLevel = DEBUG)
+
+
+from Configurables import CreateCaloCells
+createcellsBarrel = CreateCaloCells("CreateCaloCellsBarrel",
+                                    doCellCalibration=False,
+                                    addCellNoise=False, filterCellNoise=False)
+createcellsBarrel.hits.Path="ECalBarrelHits"
+createcellsBarrel.cells.Path="ECalBarrelCells"
+
```
Comment (faltovaj) [inline Detector/DetStudies/tests/options/upstreamMaterial_inclinedEcal.py]: Maybe we can use one of the simulation files already produced in other tests, not to run the simulation again. We would just need to save the particle in the simulation job I guess. Or do you run new simulations as you need higher statistics?

File: Detector/DetFCChhECalInclined/compact/FCChh_ECalBarrel_Common_upstream.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        version="1.0">
+    <comment>
+      Common settings for the inclined EM calorimeter.
+      The barrel is filled with liquid argon. Passive material includes lead in the middle and steal on the outside, glued together.
+      Passive plates are inclined by a certain angle from the radial direction.
+      In between of two passive plates there is a readout.
+      Space between the plate and readout is of trapezoidal shape and filled with liquid argon.
+      The calorimeter is divided into 0.5 cm layers, making the simulation flexible for the design studies.
+      Current study considers layer size const in eta: 2cm + 7 x 9cm.
+      Definition of sizes, visualization settings and readout are specified.
```
[the remark this answers is not in the document]
Comment (zaborowska) [reply Detector/DetFCChhECalInclined/compact/FCChh_ECalBarrel_Common_upstream.xml]: I modified the comment.

File: Detector/DetStudies/tests/options/upstreamMaterial_inclinedEcal.py (hunk 2 above)
Comment (zaborowska) [reply Detector/DetStudies/tests/options/upstreamMaterial_inclinedEcal.py]: Hi Jana, but this sim must run with the cryostat set as sensitive which in principle we could activate in other studies, but then we would need to always check if cryo==0 to filter the energy deposited in LAr.

Comment (faltovaj) [reply Detector/DetStudies/tests/options/upstreamMaterial_inclinedEcal.py]: OK, I see. I forgot about that. Sure, than it makes sense to have a new simulations for that.

**Comment by zaborowska** [comment]
retest this please

**Comment by zaborowska** [comment]
retest this please

**Pull request merged**